plugins/com.google.cloud.tools.eclipse.projectselector/src/com/google/cloud/tools/eclipse/projectselector/model/GcpProject.java

```diff
@@ -70,19 +70,19 @@ public boolean equals(Object obj) {
    * @return an AppEngine object if it has been retrieved from the backend or {@code null} if it has
    * not yet been retrieved
    */
-  public AppEngine getAppEngine() {
+  public synchronized AppEngine getAppEngine() {
```

Comment:
Good idea. Multiple concurrent threads may set this field, so I'd go with `AtomicReference`.